ibm/data_source_ibm_is_lb_listener_policies.go

```diff
@@ -0,0 +1,416 @@
+	return time.Now().UTC().String()
+}
+
+func dataSourceLoadBalancerListenerPolicyCollectionFlattenPolicies(result []vpcv1.LoadBalancerListenerPolicy) (policies []map[string]interface{}) {
+	for _, policiesItem := range result {
+		policies = append(policies, dataSourceLoadBalancerListenerPolicyCollectionPoliciesToMap(policiesItem))
+	}
+
+	return policies
+}
+
+func dataSourceLoadBalancerListenerPolicyCollectionPoliciesToMap(policiesItem vpcv1.LoadBalancerListenerPolicy) (policiesMap map[string]interface{}) {
+	policiesMap = map[string]interface{}{}
+
+	if policiesItem.Action != nil {
+		policiesMap["action"] = policiesItem.Action
+	}
+	if policiesItem.CreatedAt != nil {
+		policiesMap["created_at"] = policiesItem.CreatedAt.String()
+	}
+	if policiesItem.Href != nil {
+		policiesMap["href"] = policiesItem.Href
+	}
+	if policiesItem.ID != nil {
+		policiesMap["id"] = policiesItem.ID
+	}
+	if policiesItem.Name != nil {
+		policiesMap["name"] = policiesItem.Name
+	}
+	if policiesItem.Priority != nil {
+		policiesMap["priority"] = policiesItem.Priority
+	}
+	if policiesItem.ProvisioningStatus != nil {
+		policiesMap["provisioning_status"] = policiesItem.ProvisioningStatus
+	}
+	if policiesItem.Rules != nil {
+		rulesList := []map[string]interface{}{}
+		for _, rulesItem := range policiesItem.Rules {
+			rulesList = append(rulesList, dataSourceLoadBalancerListenerPolicyCollectionPoliciesRulesToMap(rulesItem))
+		}
+		policiesMap["rules"] = rulesList
+	}
+	if policiesItem.Target != nil {
+		targetList := []map[string]interface{}{}
+		target := policiesItem.Target.(*vpcv1.LoadBalancerListenerPolicyTarget)
+		targetMap := dataSourceLoadBalancerListenerPolicyCollectionPoliciesTargetToMap(*target)
+		targetList = append(targetList, targetMap)
+		policiesMap["target"] = targetList
+	}
+
+	return policiesMap
+}
+
+func dataSourceLoadBalancerListenerPolicyCollectionPoliciesRulesToMap(rulesItem vpcv1.LoadBalancerListenerPolicyRuleReference) (rulesMap map[string]interface{}) {
+	rulesMap = map[string]interface{}{}
+
+	if rulesItem.Deleted != nil {
+		deletedList := []map[string]interface{}{}
+		deletedMap := dataSourceLoadBalancerListenerPolicyCollectionRulesDeletedToMap(*rulesItem.Deleted)
+		deletedList = append(deletedList, deletedMap)
+		rulesMap["deleted"] = deletedList
+	}
+	if rulesItem.Href != nil {
+		rulesMap["href"] = rulesItem.Href
+	}
+	if rulesItem.ID != nil {
+		rulesMap["id"] = rulesItem.ID
+	}
+
+	return rulesMap
+}
+
+func dataSourceLoadBalancerListenerPolicyCollectionRulesDeletedToMap(deletedItem vpcv1.LoadBalancerListenerPolicyRuleReferenceDeleted) (deletedMap map[string]interface{}) {
+	deletedMap = map[string]interface{}{}
+
+	if deletedItem.MoreInfo != nil {
+		deletedMap["more_info"] = deletedItem.MoreInfo
+	}
+
+	return deletedMap
+}
+
+func dataSourceLoadBalancerListenerPolicyCollectionPoliciesTargetToMap(targetItem vpcv1.LoadBalancerListenerPolicyTarget) (targetMap map[string]interface{}) {
```

Comment:
Could we consider the oneof check!
https://pages.github.ibm.com/riaas/api-spec/spec_genesis_2021-12-07/#:~:text=LoadBalancerListenerPolicyTarget,LoadBalancerListenerPolicyTarget

Reply:
hi.. i didnt get you here